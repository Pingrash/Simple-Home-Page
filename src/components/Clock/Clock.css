.clock-container {
  flex: 1;
  margin: 1rem auto;
  text-align: center;
}

.clock-section {
  font-size: 2.5rem;
}

.date-section {
  font-size: 2rem;
}
