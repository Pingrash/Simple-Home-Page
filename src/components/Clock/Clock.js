import React from 'react';
import PropTypes from 'prop-types';
import './Clock.css';

const Clock = ({ time, date }) => {
  return (
    <div className='clock-container'>
      <div className='clock-section' data-test='time'>
        {time}
      </div>
      <div className='date-section' data-test='date'>
        {date}
      </div>
    </div>
  );
};

Clock.propTypes = {
  time: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired
};

export default Clock;
