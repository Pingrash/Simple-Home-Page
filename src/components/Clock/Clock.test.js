import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Clock from './Clock';
import { findByDataTest, checkProps } from '../../../Utilities';

// Snapshot check
it('renders correctly', () => {
  const tree = renderer.create(<Clock />).toJSON();
  expect(tree).toMatchSnapshot();
});

const setup = () => {
  const component = shallow(<Clock />);
  return component;
};

let wrapper;

beforeEach(() => (wrapper = setup()));

describe('Clock component renders correctly', () => {
  it('Should render without errors', () => {
    expect(wrapper.length).toBe(1);
  });

  it('Should render three units of time', () => {
    const time = findByDataTest(wrapper, 'time');
    expect(time.length).toBe(1);
  });

  it('Should render date', () => {
    const date = findByDataTest(wrapper, 'date');
    expect(date.length).toBe(1);
  });
});

describe('Clock component has correct props', () => {
  it('Should not have any prop errors', () => {
    const expectedProps = {
      time: 'testTime',
      date: 'testDate'
    };
    const propsError = checkProps(Clock, expectedProps);
    expect(propsError).toBeUndefined();
  });
});
