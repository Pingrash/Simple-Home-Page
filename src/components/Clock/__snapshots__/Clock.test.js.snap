// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="clock-container"
>
  <div
    className="clock-section"
    data-test="time"
  >
    00:00:00
  </div>
  <div
    className="date-section"
    data-test="date"
  >
    00/00/0000
  </div>
</div>
`;
